WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/ProductSelectorViewModelTests.swift

```diff
@@ -1404,6 +1404,70 @@ final class ProductSelectorViewModelTests: XCTestCase {
         assertEqual(bundleProduct, productToConfigure)
     }
 
+    /// We call onMultipleSelectionCompleted to save the selections made on a modal product
+    /// selector, e.g. when an iPad is in split-screen mode and is rotated to landscape.
+    /// It's a blocking operation, so we only want to do it when required.
+    func test_updateSyncApproach_calls_onMultipleSelectionCompleted_when_approach_changes_to_immediate() {
+        // Given
+        let simpleProduct = Product.fake().copy(siteID: sampleSiteID, productID: 1, purchasable: true)
+        insert(simpleProduct)
+
+        let selectedItems = waitFor { promise in
+            let viewModel = ProductSelectorViewModel(siteID: self.sampleSiteID,
+                                                     selectedItemIDs: [],
+                                                     storageManager: self.storageManager,
+                                                     syncApproach: .onButtonTap,
+                                                     onMultipleSelectionCompleted: { selectedItems in
+                promise(selectedItems)
+            })
+
+            // When
+            viewModel.addSelection(id: 1)
+            viewModel.updateSyncApproach(to: .immediate)
+        }
+
+        // Then
+        assertEqual([1], selectedItems)
```

Comment:
By checking the assertion I initially thought this was an item count, but it's the selected item IDs. Wdyt if we rename it to:
```suggestion
        assertEqual([1], selectedItemIDs)
```

Reply:
Done in 41f7880